code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -367,16 +366,13 @@
 				M.emote("gasp")
 			spawn(15)
 				M.coughedtime = 0
-
+	else
+		M.reagents.add_reagent("xeno_toxin", reagent_amount * 0.5)
 	//Topical damage (neurotoxin on exposed skin)
 	to_chat(M, "<span class='danger'>Your body is going numb, almost as if paralyzed!</span>")
-	if(prob(40 + round(amount*15))) //Highly likely to drop items due to arms/hands seizing up
-		M.drop_held_item()
-	if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-		H.temporary_slowdown = max(H.temporary_slowdown, round(effect_amt*1.5)) //One tick every two second
-
-
+	if(prob(round(reagent_amount*5))) //Likely to momentarily freeze up/fall due to arms/hands seizing up
+		M.KnockDown(0.5)
+		M.stunned += 0.5
```

Comment:
M.Stun(value).
Also these statuses update once a life cycle, so it doesn't matter making them theorically last half an unit.

Reply:
It matters for cumulative/stacking effects

Reply:
The smoke process(), and thus affect(), cycles at a similar pace compared to life(). It's very unlikely if not impossible for the effects alone to stack. 
Though you may want to use `AdjustKnockeddown()` and `AdjustStunned()` for this then.

Reply:
Umh, got it. still use those the two aforementioned procs as they instantly call update_canmove(), which makes the status kick in immediately instead of the next lifecycle.